### ISSUE TITLE
feature request: generate deterministic IDs for SVG elements to facilitate diff

### ISSUE THREAD

**Issue by gvwilson** [plotly/plotly.py#1968]
When saving figure as SVG using `write_image`, Plotly Express generates random unique IDs for elements. As a result, `git diff` thinks re-generated figures have changed even when they haven't (see screenshot). If the IDs were generated in some deterministic way depending solely on figure contents, so that they only changed when the figure itself had visually changed, `git` would be a lot happier.
![Screen Shot 2019-12-04 at 11 14 33 AM](https://user-images.githubusercontent.com/911566/70160291-4360c480-1688-11ea-9ae4-0e1e81348208.png)

**Comment by nicolaskruchten** [comment]
Thanks for creating this issue! This is something we've discussed internally with no real resolution so far... Plotly.py delegates to https://github.com/plotly/orca for SVG-generation, so we would need an Orca flag to modulate this behaviour or at least fix the random seed that it uses or something.

Heads-up @antoinerg  :)

**Comment by emmanuelle** [comment]
Great to see you here @gvwilson ! ;-) Would love to know more about how you're using plotly.py one of these days!

**Comment by gvwilson** [comment]
Thanks - I'm regenerating a lot of figures for lessons using command-line scripts and getting a linearly-increasing number of spurious diffs because of this.

**Comment by gvwilson** [comment]
@emmanuelle I'm trying to re-learn some basic statistics, mostly out of shame :-)

**Comment by emmanuelle** [comment]
Well at least it's RE-learning for you :-).

**Comment by gvwilson** [comment]
You'd be amazed how much math someone can forget in (checks) 37 years.

**Comment by antoinerg** [comment]
I agree that generating deterministic IDs should absolutely be a feature. If it requires work on Orca, I would gladly undertake it myself. However, I think it might be better to implement it directly in plotly.js. If I'm not mistaken, there might already an issue for this in https://github.com/plotly/plotly.js but I can't seem to find it at the moment (@etpinard can you point me to it?).

**Comment by etpinard** [comment]
Here -> https://github.com/plotly/orca/issues/133
